src/libs/auth.ts

```diff
@@ -97,71 +127,206 @@ const revokeTokens = async () => {
+              totalTokensUsedThisSession: authStore.get().authTokenMetadata.totalTokensUsedThisSession + 1,
+              totalTokenLoadAttemptsThisSession:
+                authStore.get().authTokenMetadata.totalTokenLoadAttemptsThisSession + 1,
+            },
+          }));
+        }
+        authStore.update((state) => ({
+          ...state,
+          authTokenMetadata: {
+            ...authStore.get().authTokenMetadata,
+            token: user.access_token,
+            createdAt: authTokenCreatedAt,
+            expiresAt: authTokenExpiresAt,
+            id: authTokenId,
+            // TODO: Change this to only update if the auth token is different than the previous token
+            totalTokensUsedThisSession: authStore.get().authTokenMetadata.totalTokensUsedThisSession + 1,
+          },
+        }));
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.success, {
+          authProvider: (user.profile as B2cIdTokenClaims).idp,
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          authTokenCreatedAt: labelTimestampMetric(authTokenCreatedAt),
+          authTokenExpiresAt: labelTimestampMetric(authTokenExpiresAt),
+          authTokenId,
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+          jwtExpiresAt: labelTimestampMetric(jwtExpiresAt),
+        });
+      },
+    ],
+    [
+      'expired',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.expired, {
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ],
+    [
+      'error',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.error, {
+          // we could potentially log the reason, but I don't know if that data is safe to log
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ]
+  );
+
+  return loadedAuthTokenState;
+};
+
+const tryLoadAuthToken = async (includeBillingScope = false, popUp = false): Promise<AuthTokenState> => {
+  const args: ExtraSigninRequestArgs = getSigninArgs(includeBillingScope);
+  const authInstance = getAuthInstance();
+  const loadedAuthTokenResponse: User | null | AuthTokenErrorState = await (popUp
```

Comment:
```suggestion
  const loadedAuthTokenResponse: User | null = await (popUp
```

These functions don't return an `AuthTokenErrorState`, that's a Terra UI type.